core/bin/zksync_server/src/main.rs

```diff
@@ -139,7 +139,12 @@ fn main() -> anyhow::Result<()> {
     {
         None => ContractsConfig::from_env_variant("GATEWAY_".to_string())
             .ok()
-            .map(Into::into),
+            .map(|value| {
+                GatewayChainConfig::from_contracts_config_and_current_settlement_layer(
+                    value,
+                    envy_load::<u64>("GATEWAY_CURRENT_SETTLEMENT_LAYER", "").unwrap(),
```

Comment:
maybe makes sense to make it part of the contracts config? I mean current settlement layer is always a property of a chain